applications/MeshMovingApplication/python_scripts/impose_mesh_motion_process.py

```diff
@@ -0,0 +1,138 @@
+import KratosMultiphysics
+import KratosMultiphysics.MeshMovingApplication as MeshMovingApplication
+
+
+def Factory(parameters, model):
+    if not isinstance(parameters, KratosMultiphysics.Parameters):
+        raise Exception(
+            'expected input shall be a Parameters object, encapsulating a json string')
+    return ImposeMeshMotionProcess(model, parameters["Parameters"])
+
+
+# Full docstring of ImposeMotionProcess (here until codacy multiline docstrings are fixed)
+#
+# Impose a rotation followed by translation on a ModelPart.
+#
+# The transformation is equivalent to:
+# 1) Translation to the reference frame (offset the origin)
+# 2) Specified rotation
+# 3) Reverse translation from the reference frame (undo origin offset)
+# 4) Specified translation
+# Note: angles in radians
+#
+# The rotation can be defined by either "euler_angles"
+# or a "rotation_axis" and "rotation_angle" pair. The following parameters can be
+# defined parametrically (see GenericFunctionUtility):
+# "euler_angles", "rotation_axis", "reference_point", "rotation_angle", "translation_vector"
+#
+# Default parameters:
+# {
+#     "model_part_name"       : "",
+#     "interval"              : [0.0, "End"],
+#     "rotation_definition"   : "rotation_axis",
+#     "euler_angles"          : [0.0, 0.0, 0.0],
+#     "rotation_axis"         : [0.0, 0.0, 1.0],
+#     "reference_point"       : [0.0, 0.0, 0.0]
+#     "rotation_angle"        : 0,
+#     "translation_vector"    : [0.0, 0.0, 0.0],
+# }
+# Note: the euler angles follow the convention specified by @ref{Quaternion} (Z, -X', Z")
+
+
+class ImposeMeshMotionProcess(KratosMultiphysics.Process):
+
+    def __init__(self,
+                 model: KratosMultiphysics.Model,
+                 parameters: KratosMultiphysics.Parameters):
+        """Impose a rotation followed by translation on a ModelPart."""
+        KratosMultiphysics.Process.__init__(self)
+
+        # 'rotation_angle' can either be a float or "End", but the validator can't handle both
+        # => convert 'rotation_angle' to the input type
+        default_parameters = self.GetDefaultParameters()
+        if parameters.Has("rotation_angle") and parameters["rotation_angle"].IsString():
+            default_parameters.RemoveValue("rotation_angle")
+            default_parameters.AddValue("rotation_angle", parameters["rotation_angle"])
+
+        parameters.ValidateAndAssignDefaults(default_parameters)
+        self.model_part = model[parameters["model_part_name"].GetString()]
+
+        # Parse interval (in lieu of a python interface for IntervalUtility)
```

Comment:
The `IntervalUtility` is heavily used in python
E.g.
https://github.com/KratosMultiphysics/Kratos/blob/8baa1c176cbda9b47406851a34555c407b085520/kratos/python_scripts/assign_flag_process.py#L47

Reply:
Thanks for the hint. I didn't find it because I was looking for its bindings in the wrong place. Apparently, it's in [```add_geometrical_utilities_to_python.py```](https://github.com/KratosMultiphysics/Kratos/blob/869241ada0defe5c5fa50408ed7924f1a0086fc6/kratos/python/add_geometrical_utilities_to_python.cpp#L285) which is a bit weird in my opinion.